### PR TITLE
Feature: updated tools for manual selection of element excitation spectral lines

### PR DESCRIPTION
**Pull request by dmgav**
The update includes:

-- Additional buttons: `Remove` button in `Fit` tab;  `Add line` and `Remove line` buttons in `Spectrum View` tab.

-- Enabling/disabling of peak selection widgets depending on data availability. `Add` and `Remove` buttons are enabled/disabled depending on whether the line highlighted in the ComboBox is in the list of selected lines.

-- Coordination of behavior of line selection tools in `Fit` and `Spectrum View` tabs. Controls from either tab can be used to add and remove lines from the list.

-- Automatic estimation of peak intensity for peak that could be added (simplified estimation as difference between raw and interpolated data). The estimated peak intensity is displayed in the `Peak Int` field of the `Fit` tab. Estimated intensity can be manually changed before the peak is added. For already selected peak, the selected peak intensity is displayed in the `Peak Int` field.

-- Multiple stability issues are fixed.

### CODE REVIEW

File: pyxrf/model/lineplot.py [hunk 1]
```diff
@@ -506,6 +747,126 @@ def show_roi_bound(self, change):
...
+                    max_line_energy, max_line_intensity = 0, 0
+                    if self.elist:
+                        for e, i in self.elist:
+                            # e - line peak energy
+                            # i - peak intensity relative to maximum peak
+                            if e >= e_fit_min and e <= e_fit_max and e > e_raw_min and e < e_raw_max:
+                                if max_line_intensity < i:
+                                    max_line_energy, max_line_intensity = e, i
+
+                    # Find the index of peak maximum in the 'fitted' data array
```
Comment (mrakitin) [inline pyxrf/model/lineplot.py]: Maybe `np.where` can be used to find what you need in the code below?

File: pyxrf/model/lineplot.py [hunk 2]
```diff
@@ -506,6 +747,126 @@ def show_roi_bound(self, change):
...
+                    n = np.clip(n, 0, len(self.data) - 1)
+                    n_raw = int(round(n))
+                    # Intensity of the fitted data at the peak
+                    in_fit = self.param_model.total_y[n_fit]
+                    # Intensity of the raw data at the peak
+                    in_raw = self.data[n_raw]
+                    # The estimated peak intensity is the difference:
+                    intensity = in_raw - in_fit
+
+                    # The following step is questionable. We assign some reasonable small number.
```
Comment (mrakitin) [inline pyxrf/model/lineplot.py]:
```suggestion
                    # The following step is questionable. We assign some reasonably small number.
```

File: pyxrf/view/lineplot.enaml [hunk 3]
```diff
@@ -108,6 +129,9 @@ enamldef PlotMain(DockItem):
                                             param_model.auto_fit_all)
                     except (AttributeError, ValueError):
                         pass
+                    if plot_model.plot_exp_opt:
+                        plot_model.plot_exp_opt = False
+                        plot_model.plot_exp_opt = True
```
Comment (mrakitin) [inline pyxrf/view/lineplot.enaml]: Why is it needed to be assigned twice? Some kind of reset?

File: requirements.txt [hunk 4]
```diff
@@ -6,7 +6,6 @@ matplotlib
 numpy
 pandas
 pillow
-pyqt5<5.11
```
Comment (mrakitin) [inline requirements.txt]: Wasn't it used before?

Comment (dmgav) [reply requirements.txt]: It is still used, but in the current configuration attempting to pip-install PyXRF into environment where PyXRF was already installed using conda messes up the environment. So currently there will be a 3-rd step in the instructions for pip-installing PyXRF:
- install scikit-beam (from nsls2forge)
- pip-install pyqt5 (if it is not installed)
- pip-install PyXRF
For develop install:
- install PyXRF from nsls2forge
- $pip install -e . (from the local PyXRF directory)
Actually only develop install is currently used and needed. Recommended way to install PyXRF is to use nsls2forge (pip install is only for experts).

File: pyxrf/view/lineplot.enaml (hunk 3 above)
Comment (dmgav) [reply pyxrf/view/lineplot.enaml]: Yes, setting it twice makes the plot to redraw no matter what the current state of '`plot_exp_opt` is. The value must change in order for the observer function to be called.

File: pyxrf/model/load_data_from_db.py [hunk 5]
```diff
@@ -681,6 +687,40 @@ def map_data2D_srx(runid, fpath,
...
+        with large data size. srx beamline only.
+    output_to_file : bool, optional
+        save data to hdf5 file if True
+
+    Returns
+    -------
+    dict of data in 2D format matching x,y scanning positions
+    """
+
+
```
[the remark this answers is not in the document]
Comment (dmgav) [reply pyxrf/model/load_data_from_db.py]: The empty lines were part of unfinished code

File: pyxrf/model/lineplot.py [hunk 6]
```diff
@@ -410,52 +470,233 @@ def plot_emission_line(self):
...
+
+    def _set_allow_add_eline(self, *, element_id, data):
+        """
+        Sets the flag, which enables/disables the controls
+        for manually adding element emission line
+
+        Parameters
+        ----------
+
+        element_id : Int
```
[the remark this answers is not in the document]
Comment (dmgav) [reply pyxrf/model/lineplot.py]: Data type from the atom library.

[the remark this answers is not in the document]
Comment (dmgav) [reply pyxrf/model/lineplot.py]: int

File: pyxrf/model/lineplot.py [hunk 7]
```diff
@@ -506,6 +747,126 @@ def show_roi_bound(self, change):
...
+
+        else:
+            if self.is_line_in_selected_list(n_id):
+                name = self.get_element_line_name_by_id(n_id)
+                intensity = self.param_model.EC.element_dict[name].maxv
+
+            else:
+
+                # Some default value
+                intensity = 1000.0
```
[the remark this answers is not in the document]
Comment (dmgav) [reply pyxrf/model/lineplot.py]: ``name`` is local: it is not used anywere else except inside conditional branches where it is defined.

File: requirements.txt (hunk 4 above)
Comment (mrakitin) [reply requirements.txt]: Sounds good.

**Pull request merged**